akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -37,7 +37,8 @@ private[remote] class EnvelopeBufferPool(maximumPayload: Int, maximumBuffers: In
     }
   }
 
-  def release(buffer: EnvelopeBuffer) = if (!availableBuffers.offer(buffer)) buffer.tryCleanDirectByteBuffer()
+  def release(buffer: EnvelopeBuffer) =
+    if (buffer.byteBuffer.isDirect && !availableBuffers.offer(buffer)) buffer.tryCleanDirectByteBuffer()
```

Comment:
Is this an unrelated fix?

Reply:
no, when duplicating it is creating a new array based ByteBuffer and those should not be released back to the pool